File: ClinicAPI/src/API/DTOs/RegisterRequest.cs
using System.ComponentModel.DataAnnotations;
using API.Models;

namespace API.DTOs;

public class RegisterRequest
{
    [Required]
    [EmailAddress]
    public string Email { get; set; } = string.Empty;

    [Required]
    [MinLength(6)]
    public string Password { get; set; } = string.Empty;

    [Required]
    public string FirstName { get; set; } = string.Empty;

    [Required]
    public string LastName { get; set; } = string.Empty;

    public string? Phone { get; set; }

    [Required]
    public UserRole Role { get; set; }

    public DateTime? DateOfBirth { get; set; }

    public string? Specialization { get; set; } // For doctors
}

File: app/src/main/java/com/example/myapplication/api/models/RegisterRequest.java
package com.example.myapplication.api.models;

public class RegisterRequest {
    private String firstName;
    private String lastName;
    private String email;
    private String password;
    private String phone;
    private int role; // 0=Patient, 1=Doctor, 2=Clerk
    private String dateOfBirth; // For Patient
    private String specialization; // For Doctor

    public RegisterRequest(String firstName, String lastName, String email, String password, String phone, int role) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.email = email;
        this.password = password;
        this.phone = phone;
        this.role = role;
    }

    // Setters for optional fields
    public void setDateOfBirth(String dateOfBirth) { this.dateOfBirth = dateOfBirth; }
    public void setSpecialization(String specialization) { this.specialization = specialization; }
}
